### PR TITLE
Add getting_started to index.rst

### PR DESCRIPTION
**Pull request by djross22**
Test, edit and add the getting_started page to the documentation

### CODE REVIEW

**Comment by djross22** [comment]
Ok, I found the .xml files in the pySBOL2 repository. I doubt they are good for SBOL3, though. 

I will try to edit the script (``pySBOL2/examples/CRISPR_example.py``) that generated the ``crisper_example.xml`` file to switch the code to pySBLO3. 
@tcmitchell - Is there any documented guidance on switching code from pySBOL2 to pySBOL3? It's not too hard to figure out, but that would be a good additional page for the docs.

The ``parts.xml`` file is only used for the ``exportToFormat`` example - which is not yet implemented. So, I don't need a replacement for that file (I was going to just comment out that section of the docs).

**Comment by djross22** [comment]
Revised plan for the example file in ``getting_started.rst``: I switched to using the `pySBOL3/test/resources/simple_library.nt` file. I didn't want to get bogged down fixing the cripsr example.

**Comment by djross22** [comment]
@tcmitchell - I've got a few questions about content on the getting_started page of the docs:
1. Is there a pySBOL3 equivalent to ``Document.append()``?
2. Is there a pySBOL3 equivalent to "Document... objects are sorted...  based on the type of object..."?
3. Can you suggest an edit for this sentence?: "SBOL-compliant URIs enable shortcuts that make the pySBOL3 API easier to use and are enabled by default. However, users are not required to use sbol-compliant URIs if they don't want to." Or should I just delete that sentence?

Thanks-

**Comment by djross22** [comment]
@tcmitchell I think the getting_started.rst doc is ready for someone else to have a look at it. 

For the last three sections on the page, I just commented everything out since those sections deal with code that is not yet written (copy/clone and format conversion) or link to one of the next doc pages (not yet edited for pySBOL3).

**Comment by tcmitchell** [comment]
> @tcmitchell - Is there any documented guidance on switching code from pySBOL2 to pySBOL3? It's not too hard to figure out, but that would be a good additional page for the docs.

There is no documented guidance for changing code from pySBOL2 to pySBOL3. I think the most important aspect of that is understanding the change from SBOL2 to SBOL3. That is the foundation of understanding pySBOL3, and thus how one might port one's code. There is a section in the SBOL3 specification about that. I think this is more of a semantics question about what it means to represent your biological objects in SBOL2 vs. SBOL3.

**Comment by tcmitchell** [comment]
> @tcmitchell - I've got a few questions about content on the getting_started page of the docs:
> 
> 1. Is there a pySBOL3 equivalent to `Document.append()`?

No there is not. `Document.append` is a way to merge multiple SBOL documents together. Perhaps that is a future enhancement to pySBOL3, and perhaps we will choose to not support such a feature. This method does not exist today.

> 2. Is there a pySBOL3 equivalent to "Document... objects are sorted...  based on the type of object..."?

No. In pySBOL3 all top level objects in a document are thrown together in a single list of objects (`Document.objects`). The separation of these top level objects into their own lists has its roots in the libSBOL, and was carried forward from pySBOL to pySBOL2. pySBOL3 stops this practice.

> 3. Can you suggest an edit for this sentence?: "SBOL-compliant URIs enable shortcuts that make the pySBOL3 API easier to use and are enabled by default. However, users are not required to use sbol-compliant URIs if they don't want to." Or should I just delete that sentence?

I think this sentence should be deleted. SBOL3 URIs are done differently from SBOL2 URIs. There are new rules governing the use and generation of URIs in SBOL3, and thus pySBOL3.

File: docs/getting_started.rst [hunk 1]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
+Setting the namespace has several advantages. It simplifies object creation and retrieval from Documents. In addition, it serves as a way for a user to claim ownership of new objects. Generally users will want to specify a namespace that corresponds to their organization's web domain.
 
 --------------------------
 Creating SBOL Data Objects
 --------------------------
 
-Biological designs can be described with SBOL data objects, including both structural and functional features.  The principle classes for describing the structure and primary sequence of a design are ComponentDefinitions, Components, Sequences, and SequenceAnnotations. The principle classes for describing the function of a design are ModuleDefinitions, Modules, Interactions, and Participations. Other classes such as Design, Build, Test, Analysis, Activity, and Plan are used for managing workflows.
+Biological designs can be described with SBOL data objects, including both structural and functional features.  
+The principle classes for describing the structure and primary sequence of a design are ``Component``, ``Sequence``, ``SequencFeature``, and ``SubComponent``.
```
Comment (jakebeal) [inline docs/getting_started.rst:102]: ``Component``, ``Sequence``, and ``Feature``.

File: docs/getting_started.rst [hunk 2]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
 
 --------------------------
 Creating SBOL Data Objects
 --------------------------
 
-Biological designs can be described with SBOL data objects, including both structural and functional features.  The principle classes for describing the structure and primary sequence of a design are ComponentDefinitions, Components, Sequences, and SequenceAnnotations. The principle classes for describing the function of a design are ModuleDefinitions, Modules, Interactions, and Participations. Other classes such as Design, Build, Test, Analysis, Activity, and Plan are used for managing workflows.
+Biological designs can be described with SBOL data objects, including both structural and functional features.  
+The principle classes for describing the structure and primary sequence of a design are ``Component``, ``Sequence``, ``SequencFeature``, and ``SubComponent``. 
+The principle classes for describing the function of a design are ``Component``, ``SubComponent``, ``Interaction``, and ``Participation``.
```
Comment (jakebeal) [inline docs/getting_started.rst:103]: ``Feature`` instead of ``SubComponent``

File: docs/getting_started.rst [hunk 3]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
+Biological designs can be described with SBOL data objects, including both structural and functional features.  
+The principle classes for describing the structure and primary sequence of a design are ``Component``, ``Sequence``, ``SequencFeature``, and ``SubComponent``. 
+The principle classes for describing the function of a design are ``Component``, ``SubComponent``, ``Interaction``, and ``Participation``. 
 
-In the official SBOL specification document, classes and their properties are represented as box diagrams. Each box represents an SBOL class and its attributes. Following is an example of the diagram for the ComponentDefinition class which will be referred to in later sections. These class diagrams follow conventions of the Unified Modeling Language.
+.. TODO: delete or edit; Is there a pySBOL3 equivalent to "Design, Build, Test, Analysis, Activity, and Plan"
+    Other classes such as Design, Build, Test, Analysis, Activity, and Plan are used for managing workflows.
```
Comment (jakebeal) [inline docs/getting_started.rst:106]: I'd suggest just leaving these out for now.

File: docs/getting_started.rst [hunk 4]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
 
 .. code:: python
 
-   >>> setHomespace('http://sbols.org/CRISPR_Example/')
-   >>> Config.setOption(ConfigOptions.SBOL_COMPLIANT_URIS, True)
-   >>> Config.setOption(ConfigOptions.SBOL_TYPED_URIS, False)
-   >>> crispr_template = ModuleDefinition('CRISPR_Template')
-   >>> print(crispr_template)
-   http://sbols.org/CRISPR_Example/CRISPR_Template/1.0.0
+    >>> target_promoter = sbol3.Component('target_promoter', sbol3.SBO_DNA)
```
Comment (jakebeal) [inline docs/getting_started.rst:131]: I'd suggest that we also add the role here for a promoter.

File: docs/getting_started.rst [hunk 5]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
 
-The following example illustrates how the URIs for ontology terms can be easily constructed, assuming they are not already part of pySBOL2's built-in ontology constants.
+The following example illustrates how the URIs for ontology terms can be easily constructed, assuming they are not already part of pySBOL3's built-in ontology constants.
```
Comment (jakebeal) [inline docs/getting_started.rst:173]: I think this example should use tyto, as is recommended in "using ontology terms"

File: docs/getting_started.rst [hunk 6]
```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
+For example, to add a promoter to a circuit design, first define the promoter and circuit as SBOL 
+``Component`` objects, then define a ``SubComponent`` as an instance of the promoter and add that 
+``SubComponent`` to the circuit's ``features`` attribute:
 
 .. code:: python
 
-    >>> point_mutation = SequenceAnnotation('PointMutation')
-    >>> target_promoter.sequenceAnnotations.add(point_mutation)
+    >>> ptet = sbol3.Component('pTetR', sbol3.SBO_DNA)
+    >>> ptet.roles = [sbol3.SO_PROMOTER]
```
Comment (jakebeal) [inline docs/getting_started.rst:325]: Set `ptet.roles` and `circuit.roles` in the constructor, as an example of good practice.

File: docs/getting_started.rst [hunk 7]
```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
 
-Or equivalently:
+Note that assigning the ``gfp_seq`` object to the ``gfp.sequences`` actually results in assignment of the object's URI. An equivalent assignment is as follows:
 
 .. code:: python
 
-   >>> point_mutation = target_promoter.sequenceAnnotations['PointMutation']
+    >>> gfp.sequences = [ gfp_seq.identity ]
+    >>> print(gfp.sequences)
+    ['http://sbolstandard.org/testfiles/GFPSequence']
```
Comment (jakebeal) [inline docs/getting_started.rst:357]: Add a example of using lookup() on an element of gfp.sequences

**Review by jakebeal — CHANGES_REQUESTED** [review]

File: docs/getting_started.rst (hunk 3 above)
Comment (bbartley) [reply docs/getting_started.rst:106]: 👍

File: docs/getting_started.rst (hunk 1 above)
Comment (tcmitchell) [inline docs/getting_started.rst:102]:
```suggestion
The principle classes for describing the structure and primary sequence of a design are ``Component``, ``Sequence``, and ``Feature``. 
```

File: docs/getting_started.rst (hunk 2 above)
Comment (tcmitchell) [reply docs/getting_started.rst:103]:
```suggestion
The principle classes for describing the function of a design are ``Component``, ``Feature``, ``Interaction``, and ``Participation``. 
```

File: docs/getting_started.rst (hunk 3 above)
Comment (tcmitchell) [reply docs/getting_started.rst:106]:
```suggestion
```

File: docs/getting_started.rst [hunk 8]
```diff
@@ -59,227 +56,207 @@ view an inventory of objects contained in the Document, simply
...
+Biological designs can be described with SBOL data objects, including both structural and functional features.  
+The principle classes for describing the structure and primary sequence of a design are ``Component``, ``Sequence``, ``SequencFeature``, and ``SubComponent``. 
+The principle classes for describing the function of a design are ``Component``, ``SubComponent``, ``Interaction``, and ``Participation``. 
 
-In the official SBOL specification document, classes and their properties are represented as box diagrams. Each box represents an SBOL class and its attributes. Following is an example of the diagram for the ComponentDefinition class which will be referred to in later sections. These class diagrams follow conventions of the Unified Modeling Language.
+.. TODO: delete or edit; Is there a pySBOL3 equivalent to "Design, Build, Test, Analysis, Activity, and Plan"
```
Comment (tcmitchell) [inline docs/getting_started.rst:105]:
```suggestion
```

File: docs/getting_started.rst (hunk 4 above)
Comment (tcmitchell) [reply docs/getting_started.rst:131]:
```suggestion
    >>> target_promoter = sbol3.Component('target_promoter', sbol3.SBO_DNA, roles=[sbol3.SO_PROMOTER])
```

File: docs/getting_started.rst [hunk 9]
```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
+(`SBOL version 3.0.0 specification document <https://sbolstandard.org/docs/SBOL3.0specification.pdf>`_). 
+For example, to add a promoter to a circuit design, first define the promoter and circuit as SBOL 
+``Component`` objects, then define a ``SubComponent`` as an instance of the promoter and add that 
+``SubComponent`` to the circuit's ``features`` attribute:
 
 .. code:: python
 
-    >>> point_mutation = SequenceAnnotation('PointMutation')
-    >>> target_promoter.sequenceAnnotations.add(point_mutation)
+    >>> ptet = sbol3.Component('pTetR', sbol3.SBO_DNA)
```
Comment (tcmitchell) [inline docs/getting_started.rst:324]:
```suggestion
    >>> ptet = sbol3.Component('pTetR', sbol3.SBO_DNA, roles=[sbol3.SO_PROMOTER])
```

File: docs/getting_started.rst (hunk 6 above)
Comment (tcmitchell) [reply docs/getting_started.rst:325]:
```suggestion
```

File: docs/getting_started.rst [hunk 10]
```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
+    >>> ptet.roles = [sbol3.SO_PROMOTER]
+    
+    >>> circuit = sbol3.Component('circuit', sbol3.SBO_DNA)
+    >>> circuit.roles.append(sbol3.SO_ENGINEERED_REGION)
```
Comment (tcmitchell) [inline docs/getting_started.rst:328]:
```suggestion
```

File: docs/getting_started.rst [hunk 11]
```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
+``SubComponent`` to the circuit's ``features`` attribute:
 
 .. code:: python
 
-    >>> point_mutation = SequenceAnnotation('PointMutation')
-    >>> target_promoter.sequenceAnnotations.add(point_mutation)
+    >>> ptet = sbol3.Component('pTetR', sbol3.SBO_DNA)
+    >>> ptet.roles = [sbol3.SO_PROMOTER]
+    
+    >>> circuit = sbol3.Component('circuit', sbol3.SBO_DNA)
```
Comment (tcmitchell) [inline docs/getting_started.rst:327]:
```suggestion
    >>> circuit = sbol3.Component('circuit', sbol3.SBO_DNA, roles=[sbol3.SO_ENGINEERED_REGION])
```

**Review by tcmitchell — CHANGES_REQUESTED** [review]

File: docs/getting_started.rst (hunk 1 above)
Comment (tcmitchell) [reply docs/getting_started.rst]: Resolved.

**Review by tcmitchell — APPROVED** [review]

File: docs/getting_started.rst [hunk 12]
```diff
@@ -59,227 +56,205 @@ view an inventory of objects contained in the Document, simply
...
 
-The following example illustrates how the URIs for ontology terms can be easily constructed, assuming they are not already part of pySBOL2's built-in ontology constants.
+The following example illustrates how the URIs for ontology terms can be easily constructed, assuming they are not already part of pySBOL3's built-in ontology constants.
 
 .. code:: python
 
-    >>> SO_ENGINEERED_FUSION_GENE = SO + '0000288'  # Sequence Ontology term
+    >>> SO_ENGINEERED_FUSION_GENE = sbol3.SO_NS + '0000288'  # Sequence Ontology term
```
Comment (tcmitchell) [inline docs/getting_started.rst:175]:
```suggestion
    >>> SO_ENGINEERED_FUSION_GENE = tyto.SO.engineered_fusion_gene
```

File: docs/getting_started.rst [hunk 13]
```diff
@@ -59,227 +56,205 @@ view an inventory of objects contained in the Document, simply
...
     >>> SO_ENGINEERED_FUSION_GENE
-    'http://identifiers.org/so/SO:0000288'
-    >>> SBO_DNA_REPLICATION = SBO + '0000204'  # Systems Biology Ontology term
+    'https://identifiers.org/SO:0000288'
+    >>> SBO_DNA_REPLICATION = sbol3.SBO_NS + '0000204'  # Systems Biology Ontology term
```
Comment (tcmitchell) [inline docs/getting_started.rst:178]:
```suggestion
    >>> SBO_DNA_REPLICATION = tyto.SBO.DNA_replication
```

**Review by tcmitchell — CHANGES_REQUESTED** [review]

**Review by tcmitchell — APPROVED** [review]

File: docs/getting_started.rst (hunk 5 above)
Comment (tcmitchell) [reply docs/getting_started.rst:171]: This is now resolved

File: docs/getting_started.rst [hunk 14]
```diff
@@ -289,258 +264,253 @@ Setting an attribute follows the ordinary convention for assigning attribute val
...
 
-Or equivalently:
+Note that assigning the ``gfp_seq`` object to the ``gfp.sequences`` actually results in assignment of the object's URI. An equivalent assignment is as follows:
 
 .. code:: python
 
-   >>> point_mutation = target_promoter.sequenceAnnotations['PointMutation']
+    >>> gfp.sequences = [ gfp_seq.identity ]
+    >>> print(gfp.sequences)
+    ['http://sbolstandard.org/testfiles/GFPSequence']
```
Comment (tcmitchell) [inline docs/getting_started.rst:353]:
```suggestion
    ['http://sbolstandard.org/testfiles/GFPSequence']
    >>> seq2 = gfp.sequences[0].lookup()
    >>> seq2 == gfp_seq
    True
```

**Review by tcmitchell — CHANGES_REQUESTED** [review]

File: docs/getting_started.rst (hunk 7 above)
Comment (tcmitchell) [reply docs/getting_started.rst:359]: Examples have been added.

**Review by tcmitchell — APPROVED** [review]

**Comment by tcmitchell** [comment]
@jakebeal I think I have incorporated all of your suggestions.

**Pull request merged**